### PR TITLE
refactor: audit, cleanup, fixes, and readability

### PR DESCRIPTION
**Pull request by deluca-mike**

### CODE REVIEW

File: contracts/src/AxelarGateway.sol [hunk 1]
```diff
@@ -168,67 +161,106 @@ abstract contract AxelarGateway is IAxelarGateway, EternalStorage {
...
+        //        if `_update` is entered within the context of _this_ implementation itself. Consider directly calling `forceUpdate`.
+        (bool success, ) =
+            newImplementation.delegatecall(abi.encodeWithSelector(IAxelarGateway.setup.selector, setupParams));
         require(success, 'SETUP_FAILED');
 
-        setAddress(KEY_IMPLEMENTATION, newVersion);
+        _setImplementation(newImplementation);
 
-        emit Updated(address(this), newVersion);
+        emit Updated(address(this), newImplementation);
```
Comment (sammy1991106) [inline contracts/src/AxelarGateway.sol:198]: Maybe let's pick up the `Upgraded` event defined in https://eips.ethereum.org/EIPS/eip-1967? `address(this)` here would be the proxy itself so it's not correct anyway.

File: contracts/src/AxelarGatewayMultisig.sol [hunk 2]
```diff
@@ -2,312 +2,473 @@
...
+        _setUint(KEY_OWNER_EPOCH, ownerEpoch);
+    }
 
-        _deployToken(name, symbol, decimals, cap);
+    function _setOwner(
+        uint256 ownerEpoch,
+        uint256 index,
+        address account
+    ) internal {
+        // AUDIT: Should have `require(account != address(0), 'ZERO_ADDR');` like Singlesig?
```
Comment (sammy1991106) [inline contracts/src/AxelarGatewayMultisig.sol:138]: sure I agree

File: contracts/src/AxelarGatewayMultisig.sol [hunk 3]
```diff
@@ -2,312 +2,473 @@
...
+        _setOwnerEpoch(++ownerEpoch);
+        _setOwners(ownerEpoch, newOwners, newThreshold);
     }
 
-    function _getOperatorsIndex() internal view returns (uint256) {
-        return getUint(KEY_OPERATORS_INDEX);
+    function transferOperatorship(address[] memory signers, bytes memory params) external onlySelf {
+        (address[] memory newOperators, uint256 newThreshold) = abi.decode(params, (address[], uint256));
+
+        // NOTE: Should this be `_areValidOwnersInEpoch` or `_areValidOperatorsInEpoch`?
```
Comment (sammy1991106) [inline contracts/src/AxelarGatewayMultisig.sol:371]: It is correct here to use `_areValidOwnersInEpoch`.

File: contracts/src/AxelarGateway.sol (hunk 1 above)
Comment (deluca-mike) [reply contracts/src/AxelarGateway.sol:198]: Only if you're ok with `proposeUpdate` being changed to `proposeUpgrade`. But I do agree on the standardized `Upgraded` event.

Comment (sammy1991106) [reply contracts/src/AxelarGateway.sol]: yes I'll do that.

**Review by sammy1991106 — APPROVED** [review]

**Pull request merged**